parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -112,341 +102,379 @@ def add_args(from_argv=False):
-    mturk_manager.soft_block_worker(worker_id)
-    onboarding_failed_workers.add(worker_id)
-
-
-def main(opt):
-    """
-    Handles setting up and running a ParlAI-MTurk task by instantiating an MTurk manager
-    and configuring it for the qa_data_collection task.
-    """
-    random.seed(opt['seed'])
-    np.random.seed(opt['seed'])
-
-    # Set the task name to be the folder name
-    opt['task'] = os.path.basename(os.path.dirname(os.path.abspath(__file__)))
-    opt['task_description'] = {
-        'num_subtasks': opt['subtasks_per_hit'],
-        'question': opt['question'],
-    }
-    # append the contents of task_config.py to the configuration
-    opt['frontend_version'] = 1
-    opt.update(opt['task_config'])
-
-    display_agent_name = 'RatingWorker'
-
-    # Set up task queue before server
-    setup_task_queue(opt)
-
-    # Instantiate an MTurkManager with the given options and a maximum number
-    # of agents per world of 1 (based on the length of mturk_agent_ids)
-    mturk_manager = StaticMTurkManager(opt=opt)
-
-    mturk_manager.setup_server(
-        task_directory_path=os.path.dirname(os.path.abspath(__file__))
-    )
-
-    mturk_manager.set_onboard_function(onboard_function=None)
 
-    try:
-        # Initialize run information
-        mturk_manager.start_new_run()
+    def __init__(self, opt):
+        random.seed(opt['seed'])
+        self.opt = opt
+
+        # add additional opt args
+        self._supplement_opt()
+
+        # class attributes
+        self.task_queue: Queue = Queue()
+        self.onboarding_tasks: List[Dict] = []
+        self.desired_tasks: List[Dict] = []
+        self.worker_data: Dict[int, Dict[int, List]] = {}
+        self.onboarding_failed_workers: Set = set()
+
+        # read in conversations data
+        self._load_conversation_data()
+
+        # setup the task queue
+        self._setup_task_queue()
+
+        # instantiate Manager
+        self.manager = StaticMTurkManager(opt=self.opt)
+
+    def _get_worker_data(self, worker_id: int) -> Dict[str, Optional[List]]:
+        """
+        Return worker data if present, else a default dict.
+        """
+        onboarding_todo = list(range(len(self.onboarding_tasks)))
+        random.shuffle(onboarding_todo)
+        return self.worker_data.get(
+            worker_id,
+            {
+                'tasks_completed': [],
+                'conversations_seen': [],
+                'onboarding_todo': onboarding_todo,
+            },
+        )
 
-        task_group_id = mturk_manager.task_group_id
-        if opt['block_on_onboarding_fail'] and opt['block_qualification'] is None:
-            opt['block_qualification'] = task_group_id
+    def _supplement_opt(self):
+        """
+        Add additional args to opt.
+
+        Useful to add relevant options after args are parsed.
+        """
+        self.opt.update(
+            {
+                'task': os.path.basename(os.path.dirname(os.path.abspath(__file__))),
```

Comment:
is this right now? Hmm..

Reply:
alas, this is an old turk hack that we've been using forever